crates/walrus-service/src/event/event_blob_downloader.rs

```diff
@@ -259,4 +220,112 @@ impl EventBlobDownloader {
 
         Ok(blobs)
     }
+
+    /// Reads the status of an event blob.
+    ///
+    /// When reading the first certified event blob, it may be the case that the event blob is
+    /// just certified, and the storage nodes may not know it yet. Therefore, upon encountering
+    /// a nonexistent blob, we will retry and wait for the blob to be certified.
+    /// For all the earlier blobs, they must have been seen by the storage nodes, since
+    /// otherwise, no new certified event blobs would be created.
+    async fn read_event_blob_status(
+        &self,
+        event_blob_id: BlobId,
+        reading_first_event_blob: bool,
+        start_time: Instant,
+    ) -> Result<BlobStatus> {
+        let blob_status = loop {
+            let blob_status = match self
+                .walrus_client
+                .get_blob_status_with_retries(&event_blob_id, &self.sui_read_client)
+                .await
+            {
+                Ok(blob_status) => blob_status,
+                Err(err) if matches!(err.kind(), ClientErrorKind::BlobIdDoesNotExist) => {
+                    BlobStatus::Nonexistent
+                }
+                Err(err) => return Err(err.into()),
+            };
+
+            if blob_status == BlobStatus::Nonexistent && reading_first_event_blob {
```

Comment:
Should we try to consolidate the retry logic in the two functions, they seem almost similar. Alternatively, what if we skip reading the status and directly read the blob for all event blobs? One way to consolidate might be: 
```
async fn retry_and_read_blob<F, Fut, T>(
    &self,
    func: F,
    blob_id: BlobId,
) -> anyhow::Result<T>
where
    F: Fn() -> Fut,
    Fut: std::future::Future<Output = anyhow::Result<T>>,
{
```
and then we pass `read_blob` and `read_blob_status`. Or maybe we can use the generic `retry_with_backoff` we may already have in our utils.

Reply:
> what if we skip reading the status and directly read the blob for all event blobs

I think that might work, although this could cause a longer timeout on the first expired blob. Then again, that is only reached anyway for newly joining nodes *once*, so the performance there is pretty irrelevant.